[SPARK-35825][INFRA] Increase the heap and stack size for Maven build

gengliangwang opened the pull request:
<!--
Thanks for sending a pull request!  Here are some tips for you:
  1. If this is your first time, please read our contributor guidelines: https://spark.apache.org/contributing.html
  2. Ensure you have added or run the appropriate tests for your PR: https://spark.apache.org/developer-tools.html
  3. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP][SPARK-XXXX] Your PR title ...'.
  4. Be sure to keep the PR description updated to reflect all changes.
  5. Please write your PR title to summarize what this PR proposes.
  6. If possible, provide a concise example to reproduce the issue for a faster review.
  7. If you want to add a new configuration, please read the guideline first for naming configurations in
     'core/src/main/scala/org/apache/spark/internal/config/ConfigEntry.scala'.
  8. If you want to add or modify an error message, please read the guideline first:
     https://spark.apache.org/error-message-guidelines.html
-->

### What changes were proposed in this pull request?
<!--
Please clarify what changes you are proposing. The purpose of this section is to outline the changes and how this PR fixes the issue. 
If possible, please consider writing useful notes for better and faster reviews in your PR. See the examples below.
  1. If you refactor some codes with changing classes, showing the class hierarchy will help reviewers.
  2. If you fix some SQL features, you can provide some references of other DBMSes.
  3. If there is design documentation, please add the link.
  4. If there is a discussion in the mailing list, please add the link.
-->
Increase memory configuration for Maven build.
Stack size: 64MB => 128MB
Initial heap size: 1024MB => 2048MB
Maximum heap size: 1024MB => 2048MB

The SBT builds are ok so let's keep the current configuration.


### Why are the changes needed?
<!--
Please clarify why the changes are needed. For instance,
  1. If you propose a new API, clarify the use case for a new API.
  2. If you fix a bug, you can clarify why it is a bug.
-->
The jenkins jobs are unstable due to the stackoverflow errors:
 https://amplab.cs.berkeley.edu/jenkins/job/spark-master-test-maven-hadoop-3.2-jdk-11/ 
https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/job/spark-master-test-maven-hadoop-2.7/2274/

### Does this PR introduce _any_ user-facing change?
<!--
Note that it means *any* user-facing change including all aspects such as the documentation fix.
If yes, please clarify the previous behavior and the change this PR proposes - provide the console output, description and/or an example to show the behavior difference if possible.
If possible, please also clarify if this is a user-facing change compared to the released Spark versions or within the unreleased branches such as master.
If no, write 'No'.
-->
No

### How was this patch tested?
<!--
If tests were added, say they were added here. Please make sure to add some test cases that check the changes thoroughly including negative and positive cases if possible.
If it was tested in a way different from regular unit tests, please clarify how you tested step by step, ideally copy and paste-able, so that other reviewers can test and check, and descendants can verify in the future.
If tests were not added, please describe why they were not added and/or why it was difficult to add.
-->
Jenkins test

gengliangwang committed:
increase jvm stack size to 128M for maven

gengliangwang commented:
cc @HyukjinKwon @LuciferYang

SparkQA commented:
**[Test build #139976 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/139976/testReport)** for PR 32961 at commit [`575fcc8`](https://github.com/apache/spark/commit/575fcc8952520e45a34dfc64cd8c495213b179b5).

gengliangwang commented:
retest this please

HyukjinKwon commented:
128 is very big .. are we sure this fixes the problem? I tried 256 to fix up AppVeyor CI but it didn't work: https://github.com/apache/spark/pull/32901

We can merge it for now and see if it works but let's probably investigate deeper next time if the issue persists.

SparkQA commented:
**[Test build #139976 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/139976/testReport)** for PR 32961 at commit [`575fcc8`](https://github.com/apache/spark/commit/575fcc8952520e45a34dfc64cd8c495213b179b5).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

gengliangwang commented:
@HyukjinKwon I think we can trigger the java 11 test with the tag `[TEST-JAVA11]` and see how it goes.

gengliangwang committed:
retrigger test

AmplabJenkins commented:
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/139976/

SparkQA commented:
**[Test build #139978 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/139978/testReport)** for PR 32961 at commit [`2517adf`](https://github.com/apache/spark/commit/2517adffb919632423140811a046ab10cb3083e7).

SparkQA commented:
Kubernetes integration test starting
URL: https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder-K8s/44502/

SparkQA commented:
**[Test build #139978 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/139978/testReport)** for PR 32961 at commit [`2517adf`](https://github.com/apache/spark/commit/2517adffb919632423140811a046ab10cb3083e7).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/139978/

LuciferYang reviewed (approved):
We can let the jdk11 compiler work normally first, and then find the root cause

SparkQA commented:
Kubernetes integration test unable to build dist.

exiting with code: 1
URL: https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder-K8s/44505/

SparkQA commented:
Kubernetes integration test status success
URL: https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder-K8s/44502/

AmplabJenkins commented:
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/44505/

AmplabJenkins commented:
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/44502/

dongjoon-hyun reviewed (commented):
Hi, @gengliangwang . Could you add an empty commit once more? In general, all CIs fail on this PR currently.
- Jenkins job fails because currently Jenkins Java11 PR Builder seems to fail at some nodes. It's a Jenkins configuration issue.
```
[error] java.io.IOException: Cannot run program "/usr/java/jdk-11.0.1/bin/java" (in directory "/home/jenkins/workspace/SparkPullRequestBuilder@2/sql/catalyst"): error=2, No such file or directory
```
- GitHub Action fails with Python UT. It will be irrelevant, but re-triggering is easier than the investigation in this case.
- AppVeyor fails with some reason.

dongjoon-hyun commented:
Could you generalize the PR description, @gengliangwang ? It seems that `java.lang.StackOverflowError` happens at Java 8 with Hadoop 2.7, too.
- https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/job/spark-master-test-maven-hadoop-2.7/2274/consoleFull

gengliangwang committed:
increase heap size as well

gengliangwang commented:
@dongjoon-hyun Thanks. I have updated the PR description.
Also, I decided to increase the heap size as well since the current 1G is soft of low. So, I created a new jira for this and updated the PR title.

SparkQA commented:
**[Test build #140019 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/140019/testReport)** for PR 32961 at commit [`a2c3f84`](https://github.com/apache/spark/commit/a2c3f84d03762609d0d821cec92b131c4516b74d).

SparkQA commented:
Kubernetes integration test starting
URL: https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder-K8s/44545/

SparkQA commented:
Kubernetes integration test status success
URL: https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder-K8s/44545/

SparkQA commented:
**[Test build #140019 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/140019/testReport)** for PR 32961 at commit [`a2c3f84`](https://github.com/apache/spark/commit/a2c3f84d03762609d0d821cec92b131c4516b74d).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/44545/

AmplabJenkins commented:
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/140019/

dongjoon-hyun commented:
For the UT failure, I made a PR.
- https://github.com/apache/spark/pull/32976

dongjoon-hyun reviewed (approved):
+1, LGTM (to stablize master branch for Apache Spark 3.2.0).
I agree with the release manager, @gengliangwang . Merged to master.
Let's see and stabilize more.

dongjoon-hyun commented:
I'm still monitoring this and unfortunately it seems to exist still.
- https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/job/spark-master-test-maven-hadoop-2.7/2278/console

dongjoon-hyun commented:
At this time, I saw it in SBT build at the latest commit.
- https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/job/spark-master-test-sbt-hadoop-3.2/2647/consoleFull

gengliangwang commented:
@dongjoon-hyun Yes it's bad..

LuciferYang commented:
Could this be a problem with Scala 2.12.x itself?  Have we ever seen the same problem with Scala 2.13.x ? I'm just guessing

LuciferYang commented:
@dongjoon-hyun It seems that the problem still exists,

```
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/140787/consoleFull
```
